src/tools/kata-ctl/src/ops/env_ops.rs

```diff
@@ -0,0 +1,445 @@
+        cpus: cores,
+    };
+
+    let memory_info = get_memory_info()?;
+
+    let guest_protection = arch_specific::available_guest_protection()?.to_string();
+
+    let support_v_socks = utils::supports_vsocks(utils::VHOST_VSOCK_DEVICE)?;
+
+    Ok(HostInfo {
+        kernel: host_kernel_version,
+        architecture: String::from(std::env::consts::ARCH),
+        distro: host_distro,
+        cpu: host_cpu,
+        memory: memory_info,
+        available_guest_protection: guest_protection,
+        vm_container_capable: true, // TODO: implement
+        support_v_socks,
+    })
+}
+
+pub fn get_runtime_info(toml_config: &TomlConfig) -> RuntimeInfo {
+    let version = VersionInfo {
+        semver: String::from(version::VERSION),
+        commit: String::from(version::COMMIT),
+    };
+
+    let config_path = TomlConfig::get_default_config_file();
+    let mut toml_path = String::new();
+    if config_path.is_ok() {
+        let p = config_path.unwrap();
+        let path_str = p.to_str();
+        toml_path = match path_str {
+            Some(s) => String::from(s),
+            None => String::new(),
+        };
+    }
+
+    RuntimeInfo {
+        // TODO: Needs to be implemented
+        path: "TBD".to_string(),
+        version,
+        experimental: toml_config.runtime.experimental.clone(),
+        guest_se_linux_label: String::from("not implemented yet"),
+        debug: toml_config.runtime.debug,
+        trace: toml_config.runtime.enable_tracing,
+        disable_guest_seccomp: toml_config.runtime.disable_guest_seccomp,
+        disable_new_net_ns: toml_config.runtime.disable_new_netns,
+        sandbox_cgroup_only: toml_config.runtime.sandbox_cgroup_only,
+        static_sandbox_resource_mgmt: toml_config.runtime.static_sandbox_resource_mgmt,
+        config: RuntimeConfigInfo { path: toml_path },
+    }
+}
+
+pub fn get_agent_info(toml_config: &TomlConfig) -> Result<AgentInfo> {
+    let agent_config = toml_config
+        .agent
+        .get(&toml_config.runtime.agent_name)
+        .ok_or("could not find agent config in configuration")
+        .map_err(|e| anyhow!(e))?;
+
+    Ok(AgentInfo {
+        debug: agent_config.debug,
+        trace: agent_config.enable_tracing,
+    })
+}
+
+use std::process::Command;
+
+pub fn get_command_version(cmd: &str) -> Result<String> {
+    // Path is empty in case of dragonball hypervisor
+    if cmd.is_empty() {
+        return Ok("unknown".to_string());
+    }
+    let output = Command::new(cmd)
+        .arg("--version")
+        .output()
+        .map_err(|e| anyhow!(e))?;
+
+    let version = String::from_utf8(output.stdout).map_err(|e| anyhow!(e))?;
+
+    Ok(version)
+}
+
+pub fn get_hypervisor_info(
+    toml_config: &TomlConfig,
+) -> Result<(HypervisorInfo, ImageInfo, KernelInfo, InitrdInfo)> {
+    let hypervisor_config = toml_config
+        .hypervisor
+        .get(&toml_config.runtime.hypervisor_name)
+        .ok_or("could not find hypervisor config in configuration")
```

Comment:
Same issue as the agent above. See: https://github.com/kata-containers/kata-containers/issues/5954